### DISCUSSION TITLE
Recommendation for Open-source log based alerting tool.

### DISCUSSION THREAD

**Discussion by karthiknagraj** [cncf/tag-observability#75] [Q&A]
Hi All,
Can anyone please suggest to me a good open-source alerting tool that can send alerts for the data in DB (Elasticsearch, MongoDB, etc)
I have a setup in which Kubernetes container logs are sent to elasticsearch and now I need some tool that can alert me on error logs and any log which I think should send out an alert. I saw one of the potential candidates for my requirement, I.e., elastalert (https://github.com/Yelp/elastalert). But this project is deprecated.

**Comment by karthiknagraj** [comment]
Hi, Anyone has any leads on this?

**Reply by vinaychandrasekhar** [reply]
If you are using Elasticsearch, you might want to check out the [alerting functionality](https://www.elastic.co/guide/en/observability/current/create-alerts.html) in Kibana for your needs.

**Comment by tonychoe** [comment]
I don't know how to generate alerts from the logs stored in ElasticSearch. But I'd recommend to check Grafana Loki. It is an excellent log grep tool and also supports powerful log-metric queries. It can generate log-metrics automatically (recording rules) and send the resulting metrics to Prometheus-compatible remote write endpoint to persist and generate alerts (alerting rules) and send the resulting alerts to Alertmanager to notify.
